devito/types/dimension.py

```diff
@@ -927,7 +942,7 @@ def _arg_defaults(self, _min=None, size=None, alias=None):
         # `factor` endpoints are legal, so we return them all. It's then
         # up to the caller to decide which one to pick upon reduction
         dim = alias or self
-        if dim._factor is None or size is None:
+        if dim.condition is not None or size is None:
```

Comment:
so we don't check _factor anymore here?

Reply:
Yeah, basically if there is a condition then you can't safely use the factor to infer arg bounds